transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/SQLiteEventStore.java

```diff
@@ -0,0 +1,443 @@
+  }
+
+  @Override
+  public void recordFailure(Iterable<PersistedEvent> events) {
+    StringBuilder query =
+        new StringBuilder("UPDATE events SET num_attempts = num_attempts + 1 WHERE _id in (");
+    Iterator<PersistedEvent> iterator = events.iterator();
+    while (iterator.hasNext()) {
+      query.append(iterator.next().getId());
+      if (iterator.hasNext()) {
+        query.append(',');
+      }
+    }
+    query.append(')');
+    inTransaction(
+        db -> {
+          db.compileStatement(query.toString()).execute();
+          db.compileStatement("DELETE FROM events WHERE num_attempts >= " + MAX_RETRIES).execute();
+          return null;
+        });
+  }
+
+  @Override
+  public void recordSuccess(Iterable<PersistedEvent> events) {
+    StringBuilder query = new StringBuilder("DELETE FROM events WHERE _id in (");
+    Iterator<PersistedEvent> iterator = events.iterator();
+    while (iterator.hasNext()) {
+      query.append(iterator.next().getId());
+      if (iterator.hasNext()) {
+        query.append(',');
+      }
+    }
+    query.append(')');
+    getDb().compileStatement(query.toString()).execute();
+  }
+
+  @Override
+  @Nullable
+  public Long getNextCallTime(String backendName) {
+    try (Cursor cursor =
+        getDb()
+            .rawQuery(
+                "SELECT next_request_ms FROM backends WHERE name = ?",
+                new String[] {backendName})) {
+      if (cursor.moveToNext()) {
+        return cursor.getLong(0);
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public boolean hasPendingEventsFor(String backendName) {
+    try (Cursor cursor =
+        getDb()
+            .rawQuery(
+                "SELECT 1 FROM events WHERE backend_id = ? LIMIT 1", new String[] {backendName})) {
+      return cursor.moveToNext();
+    }
+  }
+
+  @Override
+  public void recordNextCallTime(String backendName, long timestampMs) {
+    inTransaction(
+        db -> {
+          ContentValues values = new ContentValues();
+          values.put("next_request_ms", timestampMs);
+          int rowsUpdated = db.update("backends", values, "name = ?", new String[] {backendName});
+
+          if (rowsUpdated < 1) {
+            values.put("name", backendName);
+            db.insert("backends", null, values);
+          }
+          return null;
+        });
+  }
+
+  @Override
+  public Iterable<PersistedEvent> loadAll(String backendName) {
+    return inTransaction(
+        db -> {
+          List<PersistedEvent> events = loadEvents(db, backendName);
+          return join(events, loadMetadata(db, events));
+        });
+  }
+
+  /** Loads all events for a backend. */
+  private List<PersistedEvent> loadEvents(SQLiteDatabase db, String backendName) {
+    List<PersistedEvent> events = new ArrayList<>();
+    try (Cursor cursor =
+        db.query(
+            "events",
+            new String[] {
+              "_id", "transport_name", "priority", "timestamp_ms", "uptime_ms", "payload"
```

Comment:
I don't think there is, if we do that it would make sense to also rewrite queries to use these constants, and that would effectively make them unreadable, while not using them in queries will defeat the purpose of having constants in the first place. wdyt?